### ISSUE TITLE
Japanese language resources

### ISSUE THREAD

**Issue by coolvitto** [Mercury13/unicodia#424]
Although it is not complete, I have created a Japanese language file.
I would be happy if you could add it to the app.
[Japanese_Resources.zip](https://github.com/user-attachments/files/17954549/Japanese_Resources.zip)
- - - - - - - - -
Best regards,
by coolvitto

**Comment by Mercury13** [comment]
Oh, thank you, I’ve just had a nap under a big blackout, and woke up to appliances turning on.
Before that I managed to improve the look on Windows other than 10, and it was a real detective.
I did not expect you do so quickly, and still need to write how to translate *.utran with *.uorig.

**Comment by Mercury13** [comment]
No functionality, just a dialogue and boilerplate code.
It’ll turn *.uorig to *.utran.
![изображение](https://github.com/user-attachments/assets/c2303110-388d-437c-9142-a0f384cad10d)

**Comment by Mercury13** [comment]
P.S. These are default settings, and suitable for your deed.

**Comment by coolvitto** [comment]
Sorry, I don't understand.
*.uorig to *.utran.
Is this what I need to do?

Is "Translate with original" some kind of tool?

**Comment by Mercury13** [comment]
It’s going to be some kind of tool, under Tools menu, that will help to preserve what you have already done. I’m writing it, you forced me. Maybe tonight.

**Comment by Mercury13** [comment]
@coolvitto 
UTranslator released. ja.utran is in repo now.

If you translated more, use **Tools → Translate with → Another *.uorig**. Default settings are probably enough, except priority: you may have edited some string.

I modified locale.xml, and in those places where I’m sure I removed comment.
1. Japanese does not have plural rules.
2. Decimal point is period.
3. Native name is Nihongo.

e{1} is version name like “e12.1” (emoji 12.1). You may leave as is, or find some ideograph, IDK.

**Comment by Mercury13** [comment]
@coolvitto 
https://github.com/Mercury13/unicodia/tree/main/Lang-src

**Comment by Mercury13** [comment]
@coolvitto What’s going on now?

**Comment by coolvitto** [comment]
Is there anything I can do for you?

**Comment by Mercury13** [comment]
1. If you decided to translate — just finish. Big big thanks.
2. Switch to *.utran, it’ll just help us to work together and rule out conflicts.

**Comment by Mercury13** [comment]
@coolvitto 
1. If you decided to translate — just finish. Big big thanks.
2. Please switch to *.utran. It’ll help us to work together and rule out conflicts.

**Comment by Mercury13** [comment]
I’m working on other feature of UTranslator — “mark as bad”. I wanted to roll out first L10n’s, and did not finish this feature, as there were no languages I don’t know. Now I’ll have. When I finish, I’ll release 0.4.

**Comment by coolvitto** [comment]
ja.uorig has already been sent.

Place "en.uorig" and "ja.uorig" in the uorig folder.

**Comment by Mercury13** [comment]
@coolvitto
I’m repeating: I made ja.UTRAN, use it!!!
https://github.com/Mercury13/unicodia/tree/main/Lang-src

You have already translated something more. To port your job, use Tools → Translate with → Another *.uorig.

**Comment by coolvitto** [comment]
I asked before, but which tool is this?
The UTranslator I use doesn't have a menu called "Translate with".
![snp-1](https://github.com/user-attachments/assets/cb8176aa-c905-4231-9e03-37bf43a1da9d)

**Comment by Mercury13** [comment]
Repeating for the third time.
1. Use UTRAN!!!, not uorig.
2. Update UTranslator.

**Comment by Mercury13** [comment]
Orig = original. Tran = translation, it keeps track of what original you translated, and differentiates your comments as a translator from author’s comments.

**Comment by Mercury13** [comment]
UTranslator 0.4 with “Mark as bad” released. You use 100% and HiDPI is not for you, but I replaced pixel images with SVG.
0.3.7 with “Translate with *.uorig” was released before.

**Comment by Mercury13** [comment]
Of course, it’ll be a translation, but… empty, and you should port your work using “Translate → Another *.uorig”.
(I did not change original since then, as I worked on UTranslator.)

**Comment by coolvitto** [comment]
Hmm, I'm not good at English so I can't understand it well.

Do you mean open ja.utran in 0.4 and save it as ja(1).utran for example?

**Comment by Mercury13** [comment]
Just edit ja.utran and give it to me with any means possible (pull request, archive etc).
To preserve the work you’ve already done, you may need this
![image](https://github.com/user-attachments/assets/4d2c06ad-f383-4741-974c-e23403defcb6)
![image](https://github.com/user-attachments/assets/68016b7e-7c19-4304-9f49-5a090ca78aa7)

**Comment by Mercury13** [comment]
I invented this approach to L10n for myself, but it proved the best for translating a program that lives and is released frequently.
First in a commercial program, then recreated a n open-source translation utility from white room.

**Comment by Mercury13** [comment]
The common approach is Gettext, and it’s suitable for living program, and cool for unofficial translations, but suitable for short strings only.

**Comment by Mercury13** [comment]
What is general operation?
Even partial translation is nice.
180k characters (I counted one day) are not a cakewalk.

**Comment by coolvitto** [comment]
By "general operation" I mean the menu, the first item, etc.
However, I will try to translate as many of the other items as possible.

**Comment by Mercury13** [comment]
Then go to project settings and turn off pseudo-localization, as there will be untranslated strings and you are not going to touch them.
![image](https://github.com/user-attachments/assets/bcacbe8a-58f4-4841-8605-5fa74cfe140e)
![image](https://github.com/user-attachments/assets/3aa906a6-95df-49ba-a4ce-4d05f44aa4db)

At least translate short strings.

**Comment by Mercury13** [comment]
DO NOT translate the string you don’t want to touch. Just leave then red.

**Comment by coolvitto** [comment]
Please make sure this file is OK before proceeding with the translation.
[ja.zip](https://github.com/user-attachments/files/17986651/ja.zip)

**Comment by Mercury13** [comment]
Technically OK, I save it in the repo.
1. Check for Dating.Century.*, it’s **19th 世紀** now.
2. You’ll translate this: Prop.Kx.Data = {1} {2} {3|one=? stroke|many=? strokes}. AFAIK, Japanese has no singular and plural, and you may write {1} {2} {3} (whatever in Japanese)
3. Same for Prop.Head.NChars: just {1} 文字

**Comment by Mercury13** [comment]
For Prop.Head.NChars I’ll add another i18n fail-safe in code.

**Comment by coolvitto** [comment]
Ok, now I will proceed with the translation.

**Comment by Mercury13** [comment]
I mean: you MAY add or remove one/few/many anywhere. For strings they are just unused, for numbers they affect generation of phrases, and if you need (or don’t need) this mechanism, just remove.

**Comment by coolvitto** [comment]
I have translated everything except for Block and Script.
Please check.
![sshot-2](https://github.com/user-attachments/assets/e5121eb2-4290-4fec-b7ae-da0ff89d22e1)
I'd like to translate Block and Script as well, but I think this will take a few months.
I'm not sure if I'll be able to start on it as I have other work to do.

**Comment by Mercury13** [comment]
I may have changed something. What do you do?
Download the new en.uorig. Load JA.UTRAN. Press File → Update data.
And see what changed.

**Comment by coolvitto** [comment]
Please explain the steps in detail.
1. Open en.uorig
2. Load JA.UTRAN. (???)
I don't really understand what "Load JA.UTRAN." means.

**Comment by Mercury13** [comment]
1. DOWNLOAD the new en.uorig from repo. Put it in place of old en.uorig.
2. Open JA.UTRAN.
3. Press File → Update data.

You just check what changed.

**Comment by coolvitto** [comment]
Updated.
[JapaneseLanguageFiles_up1.zip](https://github.com/user-attachments/files/18035632/JapaneseLanguageFiles_up1.zip)

**Comment by Mercury13** [comment]
You are really cool. I’m going to release soon, and incomplete Japanese will come too.

**Comment by coolvitto** [comment]
Ok, thank you very much. 
I'm looking forward to it.

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/d9ff87e0-695a-4e50-ac8e-200bc36e9b0b)
Is it right?
1. Bulleted list is written with a 1-line gap.
2. The term “precomposed” is not translated.

**Comment by coolvitto** [comment]
In addition to the specified items, some other corrections were made.

- - - - - - - - -
Best regards,
by coolvitto
________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月8日 2:17
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


image.png (view on web)<https://github.com/user-attachments/assets/d9ff87e0-695a-4e50-ac8e-200bc36e9b0b>
Is it right?

  1.  Bulleted list is written with a 1-line gap.
  2.  The term “precomposed” is not translated.

**Comment by Mercury13** [comment]
How to write your name in About?

**Comment by coolvitto** [comment]
Please call me "coolvitto".
________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月9日 4:59
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


How to write your name in About?

**Comment by coolvitto** [comment]
And here is the homepage:
https://coolvitto.hateblo.jp

________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月9日 4:59
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


How to write your name in About?

**Comment by Mercury13** [comment]
Working on Tulu-Tigalari right now, and I wondered: syllable KR works, though I wrote nothing?
It turned out, K+R accidentally stuck together so ideally that I confused them with prebuilt ligature KR.
And KRAu was decomposed as [AK][R][u] (ideal decomposition is [A][KR][u], mine is [AKR][u]).
All one-consonant syllables done, 80% two-consonant, and a few three-consonant.

**Comment by Mercury13** [comment]
IDK what to do with syllables like CNy (C is any consonant, Ny is the largest Tulu letter).

**Comment by coolvitto** [comment]
Just to confirm, is this email being sent to me?
I'm sorry, but it's too difficult for me to understand.

coolvitto
________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月10日 1:07
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


Working on Tulu-Tigalari right now, and I wondered: syllable KR works, though I wrote nothing?
It turned out, K+R accidentally stuck together so ideally that I confused them with prebuilt ligature KR.
And KRAu was decomposed as [AK][R][u] (ideal decomposition is [A][KR][u], mine is [AKR][u]).
All one-consonant syllables done, 80% two-consonant, and a few three-consonant.

**Comment by Mercury13** [comment]
Sorry, in a hurry for half-working Tulu font I… forgot to mention you in About. May I correct next version?
Japanese released.

**Comment by coolvitto** [comment]
I have confirmed that Japanese has been added in v2.9.3.
Thank you for your quick response.

Please add more about me when you have time.

Best regards
coolvitto
________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月11日 22:53
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


Sorry, in a hurry for half-working Tulu font I… forgot to mention you in About. May I correct next version?
Japanese released.

**Comment by Mercury13** [comment]
#431 done. Thank you.

**Comment by coolvitto** [comment]
I am updating the Japanese language file, but how should I translate this?
I am getting a "Discrepancies" error.
![error](https://github.com/user-attachments/assets/6543e969-bff5-4b76-b947-e0f644f4d466)

**Comment by Mercury13** [comment]
You’ve probably found out how. Either Accept changes, or 2-click on icon.

**Comment by Mercury13** [comment]
There’s a tooltip on that icon. Actually that’s a false alarm, just fixed some typo in original language.

**Comment by coolvitto** [comment]
Ok, solved.
________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月14日 3:51
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


There’s a tooltip on that icon. Actually that’s a false alarm, just fixed some typo in original language. But tooltip days what to do.

**Comment by Mercury13** [comment]
@coolvitto Need a little help from you.
The text “colon+space” is hardcoded in many places. Is it OK for Japanese, or we need something else?

![image](https://github.com/user-attachments/assets/9c3672a8-68b4-4d31-b7da-e569aa7e7e58)
![image](https://github.com/user-attachments/assets/45ccbc2c-0d79-4dce-9b75-ed922cac283c)

**Comment by Mercury13** [comment]
And one more question. Do we need a special icon for Japanese?
![image](https://github.com/user-attachments/assets/e8273c05-7ee6-49dc-b1c5-4107357f927a)
![image](https://github.com/user-attachments/assets/fb203e9d-25e4-46b2-a958-3cdfa7d24e1a)

**Comment by Mercury13** [comment]
(I used to localize software non-professionally and internationalize professionally, so I know something)

**Comment by coolvitto** [comment]
The colon and space are not strictly necessary in Japanese.
However, I like to put spaces before and after the colon for readability.

When it comes to icons, I don't think you need to be particularly conscious of Japanese.
________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月15日 11:35
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


(I used to localize software non-professionally and internationalize professionally, so I know something)

**Comment by Mercury13** [comment]
> When it comes to icons, I don't think you need to be particularly conscious of Japanese.
But you’ll probably translate names of scripts and blocks. And you’ll need to draw another icon — e.g. A to WO, for simplicity in Katakana.

**Comment by Mercury13** [comment]
@coolvitto 

> When it comes to icons, I don't think you need to be particularly conscious of Japanese.

But you’ll probably translate names of scripts and blocks. And you’ll need to draw another icon — e.g. A→WO, for clarity in Katakana.

**Comment by coolvitto** [comment]
I think it's more user-friendly to translate scripts and blocks that are specific to Japanese.
(However, I don't think it's that important)

As for icons, I don't think it's important to take the trouble to create them.
(However, if you don't mind taking the time and effort to create them, I think it's more user-friendly...)
________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月15日 23:16
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


@coolvitto<https://github.com/coolvitto>

When it comes to icons, I don't think you need to be particularly conscious of Japanese.

But you’ll probably translate names of scripts and blocks. And you’ll need to draw another icon — e.g. A→WO, for clarity in Katakana.

**Comment by Mercury13** [comment]
Understand. If most names will remain in Latin → leave A→Z. Hope you’ll figure out how to insert kana to alphabetic sorting.

**Comment by Mercury13** [comment]
:) You’ve seen, the icon for technical name (not eligible for L10n until we translate to RtL, e.g. Hebrew) consists of segmented A and pixel Z. Segmented and pixel — what’s more techy?

**Comment by coolvitto** [comment]
If the text contains a mixture of alphabets and kana characters, translating the kana characters will make it more difficult to understand, so I think it's fine to ignore the kana characters.
________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月15日 23:38
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


Understand. If most names will remain in Latin → leave A→Z. Hope you’ll figure out how to insert kana to alphabetic sorting.

**Comment by Mercury13** [comment]
@coolvitto 
Extra space in Coptic is accidental, do not repeat it!

**Comment by coolvitto** [comment]
Sorry, I don't have any knowledge on the technical aspects.
________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月15日 23:44
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


:) You’ve seen, the icon for technical name (not eligible for L10n until we translate to RtL, e.g. Hebrew) consists of segmented A and pixel Z. Segmented and pixel — what’s more techy?

**Comment by Mercury13** [comment]
> If the text contains a mixture of alphabets and kana characters, translating the kana characters will make it more difficult to understand, so I think it's fine to ignore the kana characters.

I mean alphabetic sorting: Latin and kana will co-exist, and need to be sorted somehow, right?

**Comment by coolvitto** [comment]
I think it's okay to rearrange kana characters using the romanized notation.
(Example: ⒚ -> KANA)
________________________________
差出人: Mercury13 ***@***.***>
送信日r: 2024年12月15日 23:54
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


If the text contains a mixture of alphabets and kana characters, translating the kana characters will make it more difficult to understand, so I think it's fine to ignore the kana characters.

I mean alphabetic sorting: Latin and kana will co-exist, and need to be sorted somehow, right?

―

**Comment by Mercury13** [comment]
I mean, how to sort together kana and Latin.

**Comment by Mercury13** [comment]
Latin KA < Kana KA < Latin KI < Kana KI < Latin SA < Kana SA?
Latin KA < Latin KI < Kana KA < Kana KI < Latin SA < Kana SA?
All Latin < all Kana?

**Comment by Mercury13** [comment]
UTranslator updated to 0.4.1. Nothing really useful for you, but a minor bugfix, a minor change and a new feature for me.

**Comment by coolvitto** [comment]
I think

Kana KA < Kana KI < Kana SA < Latin KA < Latin KI < Latin SA

Probably...???
________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月17日 6:24
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


Latin KA < Kana KA < Latin KI < Kana KI < Latin SA < Kana SA?
Latin KA < Latin KI < Kana KA < Kana KI < Latin SA < Kana SA?
All Latin < all Kana?

**Comment by coolvitto** [comment]
thanks
________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月17日 11:01
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


UTranslator updated to 0.4.1. Nothing really useful for you, but a minor bugfix, a minor change and a new feature for me.

**Comment by Mercury13** [comment]
All kana (Katakana and Hiragana mixed) < all Latin, right? This way we’ll surely make.

**Comment by coolvitto** [comment]
Personally, that's what I think.
________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月17日 11:25
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


All kana (Katakana and Hiragana mixed) < all Latin, right? This way we’ll surely make.

**Comment by coolvitto** [comment]
I didn't want to end the year with things half-done, so I translated the whole thing.

![sshot-20](https://github.com/user-attachments/assets/464c2ba1-cb63-4a9a-8a5b-e5a4561303bc)
[Japanese_Language_Resources_24241218.zip](https://github.com/user-attachments/files/18175905/Japanese_Language_Resources_24241218.zip)

**Comment by Mercury13** [comment]
Sorry, I didn’t notice in time.
Four strings to go:
1. Sunuwar extended
2. Garay extended
3. Antarctica new
4. One string in search substitution instead of concatenation

**Comment by Mercury13** [comment]
The only sortable is CJK. I’m sorting Hira A < Kana A < Hira I < Kana I < … < Kana Wo < Latin?

**Comment by Mercury13** [comment]
Could you retranslate “基本ラテン文字 (ASCII)” to put Latin in front?

**Comment by coolvitto** [comment]
Fixed
![sshot-24](https://github.com/user-attachments/assets/65a34391-ab83-454f-bdd7-82f0062cb0a8)

**Comment by coolvitto** [comment]
As for sorting CJK, I think it's fine to simply sort the romanized version alphabetically.
Taking into account the native language makes things complicated.

**Comment by coolvitto** [comment]
Updated
[Japanese_Update20241219.zip](https://github.com/user-attachments/files/18193052/Japanese_Update20241219.zip)

**Comment by Mercury13** [comment]
I’m making this thing, automatically for now.
Was originally for sorting of Ol Chiki < Ol Onal < Old Klingon, but came in handy for Japanese.
![image](https://github.com/user-attachments/assets/99c09d75-9a15-494a-8a61-6e712d38be3a)

**Comment by Mercury13** [comment]
Katakana and Hiragana are equivalent for now, but KA<GA<KI.

**Comment by coolvitto** [comment]
Awesome!! 
I think that's fine.

**Comment by Mercury13** [comment]
You spoiled sample in block 0500 Cyr supplement.

**Comment by Mercury13** [comment]
Is this way of writing centuries OK?
![image](https://github.com/user-attachments/assets/162deef5-e63b-46c3-b78d-7530a2da65a1)

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/b15aba0e-a909-42f0-a2c6-95c13c9b24bc)

**Comment by Mercury13** [comment]
You forced me to draw the icon of gojuon, right now I’m sticking to this.
Tonight.
![image](https://github.com/user-attachments/assets/062d7bf4-395c-445a-a250-92a3e7c25cae)

**Comment by coolvitto** [comment]
All translations are translated like that.
![sshot-25](https://github.com/user-attachments/assets/62d11f08-3608-4c7c-ad44-623613bbb6dc)

Is there anything I can help you with in creating icons?

**Comment by Mercury13** [comment]
Have you ever edited Wikipedia? This markup is out there, except I did not make short links.
Even it marks links red if does not find target, like in Wikipedia.
Arguable (e.g. no consensus) probably exists in Japanese.

**Comment by Mercury13** [comment]
{{-}} = ordinary Western dash with all surrounding spaces.

**Comment by coolvitto** [comment]
I have never created a Wikipedia.

**Comment by Mercury13** [comment]
I mean, is this design OK? Because of low-fi icons, I prefer Kana to Hira.

**Comment by Mercury13** [comment]
You see the design.
![image](https://github.com/user-attachments/assets/ba20e94b-fd63-4c28-b876-a6221792dd02)
A to WO, in Kana (too lo-fi for Hira)

**Comment by Mercury13** [comment]
I’ll make it in vector tonight, probably using Noto font.

**Comment by coolvitto** [comment]
Shall I create one at the same time?
Do I have to use Noto font?

**Comment by coolvitto** [comment]
ok, I understand

**Comment by Mercury13** [comment]
You better say: does it look like A-Z in Latin ок А-Я in most types of Cyrillic?

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/4db9c329-748f-423d-a14b-0586a0d45d22)
OK?

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/acdd051b-1cee-4e36-ab53-67daeae46ac6)
![image](https://github.com/user-attachments/assets/a0be328a-f020-4222-8ea2-444671519af6)

**Comment by coolvitto** [comment]
Wow!! 
Very wonderful

**Comment by coolvitto** [comment]
I've reviewed the translation and made some changes.

![sshot-26](https://github.com/user-attachments/assets/18e5640f-41e8-4731-bd32-7f09531e4a60)

統合カナダ先住民音節 - > カナダ先住民音節
etc ...

[Japanese_Fix_20241220.zip](https://github.com/user-attachments/files/18207387/Japanese_Fix_20241220.zip)

**Comment by coolvitto** [comment]
I've uploaded some fixes.
https://github.com/Mercury13/unicodia/issues/424
[https://opengraph.githubassets.com/f5472cbc4a4824a1a8849ee6e02723717b1d4b6c4fb3e2b3878abb7c4c03cc88/Mercury13/unicodia/issues/424]<https://github.com/Mercury13/unicodia/issues/424>
Japanese language resources · Issue #424 · Mercury13/unicodia<https://github.com/Mercury13/unicodia/issues/424>
Although it is not complete, I have created a Japanese language file. I would be happy if you could add it to the app. Japanese_Resources.zip Best regards, by coolvitto
github.com


________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2024年12月20日 5:54
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


image.png (view on web)<https://github.com/user-attachments/assets/acdd051b-1cee-4e36-ab53-67daeae46ac6>
image.png (view on web)<https://github.com/user-attachments/assets/a0be328a-f020-4222-8ea2-444671519af6>

**Comment by Mercury13** [comment]
I asked about “arguable” — I mean, “no consensus about script type”.
(Sorry about black stripe — that’s peculiarities of Firefox and Paint.NET.)
I’ll check — I came home very late, and fell asleep with lights and computer on, what a luck there was no blackout and UPS did not exhaust.
![image](https://github.com/user-attachments/assets/e5a6fa9e-1b1e-495c-9ff4-495d5570eb3a)

**Comment by Mercury13** [comment]
About black stripe — that’s a trouble in translucent DIB in clipboard, even Windows itself did not make it right.
Other software (e.g. Chromoids) just ignore that format and use earlier opaque DIB. I experimented and even gave Firefox a few options.

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/0bc66b3e-4ed6-40ba-b61c-5fee7b3243f0)
You broke L10n here, the correct thing is “{1} xxx”.

**Comment by Mercury13** [comment]
You don’t need to use one/few/many, if Japanese has no plural.

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/70ffbf0e-3f26-4ca6-aadb-dfeca1aa8d1d)
Not a mistake, but could you put true apostrophe

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/70ffbf0e-3f26-4ca6-aadb-dfeca1aa8d1d)
Not a mistake, but could you put true apostrophe U+2019 here?

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/0e95d187-cbf8-42ea-987b-ea676daf5bf6)
Could you change this A→Z to Katakana?
As you don’t use a version control system, we’re forced to communicate this way.

**Comment by Mercury13** [comment]
The last A→Z should also have no space, just for uniformity.

**Comment by Mercury13** [comment]
Telugu: are you right?
![image](https://github.com/user-attachments/assets/cc14f153-06c8-46eb-9861-c657d9a07954)

**Comment by Mercury13** [comment]
Malayalam: same
![image](https://github.com/user-attachments/assets/df081aa8-2d71-431e-b01e-9c58982e4cab)

**Comment by Mercury13** [comment]
Myanmar: translate “year”
![image](https://github.com/user-attachments/assets/cce18ae6-1df9-4b2b-882f-50625a45567f)

**Comment by Mercury13** [comment]
Well, need to run

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/4f346ef3-e13f-4944-b8bf-8f4203d5e7c0)
Centuries untranslated, though I see somewhere that you use just 1…21.

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/888774fc-e899-46f4-a08f-dd2ac9c97080)
Same: {1} {2} {3} XXX

**Comment by coolvitto** [comment]
Actually, I don't really understand what this means.
Where is this message displayed?

**Comment by Mercury13** [comment]
Something of this I can fix for myself. But as you don’t use version control system, we just cannot merge our works!

**Comment by Mercury13** [comment]
As I use and you don’t, you will just overwrite my work.

**Comment by coolvitto** [comment]
I see, that's what it was.
I understand now.

**Comment by Mercury13** [comment]
Centuries were implemented in March 2022 in evacuation, as an attempt to make English, and I did not want to make true plural rules. Moreover, Russian mostly uses Roman numerals for centuries, and English does Arabic ordinals.
OK, but better just
{1} {2} {3} 画

**Comment by coolvitto** [comment]
I see.
Since Japanese doesn't have a plural form, it simply says "画".

But that doesn't match the original, so for now I'll translate it as "{1} {2} {3|one=? 画|many=? 画}".

**Comment by Mercury13** [comment]
The FIRST should be kana (localized names), the LAST always A→Z (technical names)

**Comment by Mercury13** [comment]
If you write in the first “gojuon” instead of “alphabet”, it’d be OK, as we are really striving to sort them by Japanese pronunciation.

**Comment by Mercury13** [comment]
IDK where to put a build for you.

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/590a0df2-0f97-4683-91b0-ec9c57ad34b7)

**Comment by coolvitto** [comment]
Fixed.
![名称未設定-1](https://github.com/user-attachments/assets/713f7ec4-d847-4bd0-8c8a-2973b8132190)

I'm very grateful for the various improvements you've made to support Japanese.
I'm sure many Japanese people will be very happy.

Wouldn't it be better to just put the build after the version number?

**Comment by coolvitto** [comment]
If you want to sort in Japanese, won't you have to use a database?

In that case, won't you have to create a new program???

If the search function is solid, I don't think localized sorting is that important.

Anyway, please consider not pushing yourself too hard.

**Comment by coolvitto** [comment]
This is the revised version with the corrections pointed out and other changes.

[Japanese_Fix_20241221.zip](https://github.com/user-attachments/files/18214595/Japanese_Fix_20241221.zip)

**Comment by Mercury13** [comment]
We don’t search in Japanese for now. Even in Russian it’s big hard, just broke teeth.
To sort those 300, we rely on fixup database, you’ve seen a piece. And you can even improve those fixups.
The base was intended to fix Ol Chiki < Old Klingon < Ol Onal discrepancy, but I decided to put Old after Hungarian, Turkic etc. Only for Shavian in Russian/Ukrainian I did not found out how to write it.
After a training, I’ll check.

**Comment by Mercury13** [comment]
@coolvitto Sorry, the link is broken

**Comment by Mercury13** [comment]
NChars to {1} 文字
Kangxi to {1} {2} {3} 画

**Comment by Mercury13** [comment]
Non-critical. You Japanese probably know what’s Kanbun. Do you need this long description?
Kanbun (hints to medieval Japanese) = 漢文用記号 (中世日本語のヒント)

**Comment by Mercury13** [comment]
Non-critical. IDK your attitude to foreign names, but is it right, or need Latin “3D World Atlas”?
![image](https://github.com/user-attachments/assets/ad4bd364-16ee-4d34-bbea-9cfeb5c678c5)

**Comment by Mercury13** [comment]
Non-critical, but was specifically fought in other L10n’s of Unicodia
![image](https://github.com/user-attachments/assets/2096c7c2-412f-4749-aa31-ecf0043f7e9a)
Old in the beginning of those blocks hinder search. Blocks only, in Scripts no need.

**Comment by Mercury13** [comment]
…Old Italic too, just did not got to it!

**Comment by Mercury13** [comment]
Old Permic is KoPeRuMu, so Old Permic just moved to the bottom under Coptic which is probably KoPuTu :) It’s too…

**Comment by Mercury13** [comment]
For Greek and Ancient Symbols — NO NEED, they are automatically stuck to Greek, and if it’s weirdly sorted, just prepend ellipsis!

**Comment by Mercury13** [comment]
Same for Old Arabic North and South.
REPEAT: in Blocks only, in Scripts you may write everything you want
![image](https://github.com/user-attachments/assets/ca31e47e-f7f2-47bf-a588-55edc5be671c)

**Comment by Mercury13** [comment]
REPEAT: No need: in Scripts (they are not subject to sorting), in Ancient Greek and Ancient symbols (automatically stuck to Greek, again write whatever you want)

**Comment by Mercury13** [comment]
Ancient symbols 10190 — NO NEED to touch

**Comment by Mercury13** [comment]
Here’s our new sorting order? Is seemingly OK?
![image](https://github.com/user-attachments/assets/52be4974-257b-4707-981a-fd848c7ab000)

**Comment by coolvitto** [comment]
This is what it looks like in my view.
![名称未設定-2](https://github.com/user-attachments/assets/774fec69-ebe8-489d-bea5-2106704ef062)

How about this?
Ancient symbols (古代記号)

I check the whole block again, including "Old".

**Comment by coolvitto** [comment]
For now, this is the revised version.

[Japanese_Fix_20241221-1.zip](https://github.com/user-attachments/files/18217150/Japanese_Fix_20241221-1.zip)

**Comment by Mercury13** [comment]
1. I changed something in code, first of all ease of building alphabeti

**Comment by Mercury13** [comment]
1. There are many changes in code: two L10n features work together now (did not before), easier to build complex alphabets (Russian/Ukrainian benefit too!), four more pairs stick together.
2. Different sort orders: yours is continent + gojuon (2nd, default), mine is just gojuon (1st). Well, the 2nd is default for the reason: you often know where is the script from.
3. Ancient symbols DO NOT CHANGE, I repeated.
4. NEED: Arabian, Permic, Orkhon, Italic, Hungarian, Hungarian.

**Comment by coolvitto** [comment]
Fixed

[Japanese_Fix_20241221-2.zip](https://github.com/user-attachments/files/18217842/Japanese_Fix_20241221-2.zip)

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/6ee8b4f1-46d4-43da-9994-1001169012b4)
Revert/rewrite: approximately

![image](https://github.com/user-attachments/assets/a55d81cf-919b-44fb-b8ad-802415f1f895)
REPEATING FOR THE FOURTH TIME: use red version

![image](https://github.com/user-attachments/assets/bd996311-ca89-43ef-bc8d-7d57879072ef)
Same

![image](https://github.com/user-attachments/assets/bc68ebaa-229e-46f9-b0ee-a26e130c9b84)
REPEAT: programmatically stuck together: revert

![image](https://github.com/user-attachments/assets/aac7b396-709f-4d39-8084-967329234f1a)
REPEAT: these (ancient Greek/Roman mixed) are not a concern. Revert/retranslate.

![image](https://github.com/user-attachments/assets/49bbfa59-fa57-46b3-ac0a-c6a60ecf4ac4)
Thank you, but two parentheses — is it right?

![image](https://github.com/user-attachments/assets/0e2730bc-8ebf-48ca-9870-b3b94ceea473)
Same

![image](https://github.com/user-attachments/assets/7b77791e-89c3-4d5f-bf0e-b996e3250fff)
Revert, automatically stuck to Sodgian

![image](https://github.com/user-attachments/assets/010968cc-f669-4eb3-89db-9d86431b8ecc)
Could you rewrite to put Gondi in front?

![image](https://github.com/user-attachments/assets/0349c8c1-b57e-4063-a566-01744e9f6ea8)
Revert, automatically stuck to Greek

The rest — thank you.

**Comment by Mercury13** [comment]
I’ll make a release, no critical problems left. You’ll see my programmatic L10n features.

**Comment by coolvitto** [comment]
It seems I thought too much and the translation came out strange.
I've corrected it again.

[Japanese_Fix_20241221-3.zip](https://github.com/user-attachments/files/18218229/Japanese_Fix_20241221-3.zip)

**Comment by Mercury13** [comment]
I did not sort vocabulary for now, I’ll invent some workaround. And release in the evening.
And check the rest: though I don’t know language, I test everything.

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/0594b480-54d2-4984-af86-4911153a77b3)
Red links, check!

**Comment by Mercury13** [comment]
I internationalized software professionally and localized unprofessionally :)

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/1dde2447-1615-4734-afac-6d9e427992c1)
Non-critical: extra space

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/56dfab9c-ff5c-4092-bc30-a10c64c4cf43)
Masaram OK. Where’s Gunjala?

**Comment by Mercury13** [comment]
Italic NEED old in the end, reverting
![image](https://github.com/user-attachments/assets/e7af7094-efa5-4d46-8078-4e9c37b939dd)

**Comment by Mercury13** [comment]
Permic/Persian NEED old in the end, reverting
![image](https://github.com/user-attachments/assets/6ae0d6d5-d10b-41c7-a2c9-800c43d108cd)

**Comment by Mercury13** [comment]
Arabic NEED old in the end, reverting
![image](https://github.com/user-attachments/assets/7a2a2656-135b-44d8-9c2f-ac342936ae14)

**Comment by Mercury13** [comment]
Hungarian NEED old in the end, reverting
![image](https://github.com/user-attachments/assets/40c2f7e7-fbb3-4897-8abc-f6fa1e8d09e0)

**Comment by Mercury13** [comment]
Uyghur NEED old in the end, reverting
![image](https://github.com/user-attachments/assets/b743b940-a1a6-4886-8159-48a3c421b868)

Sorry, it’s really complex just because I tried to invent some order for 3 years. Everything started from 00→FF, but 300 blocks are a royal pain to find in. If I could stick ancient Greek to Greek — cool. Cannot — invent.

The main thing is: whether “old” is main word (Ancient symbols), auxiliary block (Old Sogdian, Ancient Greek) or an adjective to something more important (Old Italic, Old Persian). Depending on this question, the answer is: 1) Better to front. 2) No matter, we’ll find anyway. 3) Better to back.

Thank you.

**Comment by Mercury13** [comment]
I’ll write about “Old” in readme. You made me articulate the rule I invented for these three years.

**Comment by Mercury13** [comment]
I wrote this way.

Is **Old** in the front or in the back? It depends. 1) In Scripts — as convenient. In Blocks… 2) Old is the main word (Ancient symbols) → better front. 3) Auxiliary block (Old Sogdian, Ancient Greek) → no matter, we’ll find it anyway by looking around Greek. 4) Old is an adjective to something more important (Italic old, Mongolian old, Permic old) → better back. It’s just ease of finding a block in the long list of 300 blocks.

**Comment by coolvitto** [comment]
I apologize for having to point things out repeatedly due to my poor understanding of English.

Revised version v5
[Japanese_Fix_20241221-5.zip](https://github.com/user-attachments/files/18218566/Japanese_Fix_20241221-5.zip)

The "Old" issue is a very troubling one.

**Comment by Mercury13** [comment]
As you see: white — international, black — technical, blue — Europe, yellow — Asia, pink — CJK, green — Indian and Pacific Ocean, brown — Africa, red — America.

**Comment by Mercury13** [comment]
If someone adds fictional scripts at last, they’ll be violet.

**Comment by Mercury13** [comment]
Maybe I’ll separate RtL Asia to more brownish-yellow.

**Comment by coolvitto** [comment]
>international, black — technical, blue — Europe, yellow — Asia, pink — CJK, green — Indian and Pacific Ocean, 
>brown — Africa, red — America (including NP Hmong invented in America)

Oh! That's what it meant!

**Comment by Mercury13** [comment]
Source: common but untrue interpretation of Olympic rings. Pink is from sakura, green on black is from The Matrix.

**Comment by Mercury13** [comment]
Myanmar
![image](https://github.com/user-attachments/assets/33dc1f6c-693a-4f31-8381-1c24e3667c9d)

**Comment by Mercury13** [comment]
Telugu
![image](https://github.com/user-attachments/assets/cc4c42da-82ab-4e75-a799-84c7d9f6083d)

**Comment by Mercury13** [comment]
Arabic A (08A0) — strange Latin things
![image](https://github.com/user-attachments/assets/5098758d-bea7-4309-92e4-46c4d9ed24ab)

**Comment by Mercury13** [comment]
Syriac ex — right?
![image](https://github.com/user-attachments/assets/3fdbf92f-8e80-41a7-bd19-30d4973e1b0a)

**Comment by Mercury13** [comment]
Syriac — probably Japanese here
![image](https://github.com/user-attachments/assets/c1697258-a5e9-479b-86f4-56dfbc66109b)

**Comment by Mercury13** [comment]
Cyrillic ex (0500) — wrong sample
![image](https://github.com/user-attachments/assets/ec874390-d3f4-4405-9576-0430ea0eb227)

**Comment by Mercury13** [comment]
IPA 0250 — should be Latin “laurel” somewhere. What’s the story? English word “laurel” was pronounced with an opera bass. It came through bad speaker and bad microphone, and who had this bad speaker, heard “Yanny” — and people with at least earphones continued to hear “laurel”.
![image](https://github.com/user-attachments/assets/2447fb89-addd-4ce5-8df8-4f96f134149a)

**Comment by Mercury13** [comment]
Mongol — clearly Japanese here
![image](https://github.com/user-attachments/assets/e52b5a36-f36e-40b8-a9fa-4960f19d738d)

**Comment by Mercury13** [comment]
19E0 Khmer ex — is this right?
![image](https://github.com/user-attachments/assets/a8d4cb52-75ef-4ab2-96b7-49dc6ed1a3ed)

**Comment by Mercury13** [comment]
Batak — virama may be either in Japanese or in Latin, does not really matter, but this is surely Latin BA.
![image](https://github.com/user-attachments/assets/bd07674d-5b40-4be0-8d14-a9ce6dff0fe2)

**Comment by Mercury13** [comment]
1C90 Georgian Mtavruli — ways too much Latin, smell wrong but cannot prove as I don’t know language
![image](https://github.com/user-attachments/assets/16033802-6f6a-4d80-8474-af150b7485d6)

**Comment by Mercury13** [comment]
1F00 Greek ex — let specific things be Latin, but well-known umlauts — either all Latin, or all Japanese
![image](https://github.com/user-attachments/assets/d9782f62-8b0e-458f-80dc-2c3d6e50dfc4)

**Comment by Mercury13** [comment]
2000 punctuation. You’ve put a native Japanese word here, and Latin origin becomes irrelevant.
![image](https://github.com/user-attachments/assets/5bffc101-4e59-4096-9ed6-fb3d45a16040)

**Comment by Mercury13** [comment]
Term **Emoji**: you are Japanese, probably rewrite
![image](https://github.com/user-attachments/assets/d859974a-6c78-4847-b209-1f95a4eb7016)

**Comment by Mercury13** [comment]
Term Emoji: you are Japanese, probably rewrite
![image](https://github.com/user-attachments/assets/82fd6944-0d36-48fd-9c02-844c33a64880)

**Comment by Mercury13** [comment]
I translated Unicodia to Ukrainian in two months (Dec 2022 to Jan 2023, together with other work — HiDPI), and it became on par by frequency of mistakes in a year. About English — I don’t remember :(

**Comment by Mercury13** [comment]
May I stop for now? :)

**Comment by Mercury13** [comment]
2700 Dingbats. IDK whether this word is Japanese or Latin, both are valid.
![image](https://github.com/user-attachments/assets/935421a6-1530-4ca6-b0ed-7cf3b14b48cb)

**Comment by Mercury13** [comment]
Prop.Type — probably more words in Japanese, at least Arguable and Codes and shorthand
![image](https://github.com/user-attachments/assets/d7be264c-0fef-4812-aeaf-dc0142997f2d)

**Comment by Mercury13** [comment]
2B00 misc arrows — unclosed quote
![image](https://github.com/user-attachments/assets/6a662381-0ac4-4fa2-b058-3415b789f8e0)

**Comment by Mercury13** [comment]
I don’t know Japanese but check how Japanese sticks together with Latin.

**Comment by Mercury13** [comment]
https://github.com/Mercury13/unicodia/releases/tag/v2.9.3
Download pre-release 2.9.3.1, many of your work there

**Comment by coolvitto** [comment]
>May I stop for now? :)

No, please continue.
If you notice anything strange, please feel free to point it out.

I translate various apps, and I'm happy that I can increase my knowledge by translating.

However, it's the end of the year and I'm free, but I may not be able to keep up when my actual work gets busy.

This is a revised version of what I've done so far.
[Japanese_Fix_2024_12_22.zip](https://github.com/user-attachments/files/18219510/Japanese_Fix_2024_12_22.zip)

**Comment by Mercury13** [comment]
Please take new en.uorig: after I implemented “sorting telltales”, I found six technical mistakes even in English.

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/ff0e0a37-bc35-4d59-8a0f-d592a5c41116)
Isn’t it better to write Hellschreiber in Latin? — IDK how Japanese put together Latin and Katakana, see the last character 2BFF
![image](https://github.com/user-attachments/assets/75cddc2a-eb04-442c-808c-58b5929348a8)

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/b57a4d52-eb3a-4c7d-9eff-1d66692db5d5)
You still got Department of Redundancy Department. Specially for Japanese you should really simplify.

**Comment by Mercury13** [comment]
@coolvitto Do you like sorting in our pre-release 2.9.3.1?

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/72e3eae9-33cc-4518-9e26-ac736b251389)
Got term in Japanese

**Comment by Mercury13** [comment]
2D80 Ethiopic ex
Incomplete link
![image](https://github.com/user-attachments/assets/910ccf13-bb5e-40de-8c11-a951bf08fadb)

**Comment by Mercury13** [comment]
IDC 2FF0: sure you’ve got this name in Japanese

![image](https://github.com/user-attachments/assets/80291586-1e86-412e-9936-feaf23401276)

**Comment by Mercury13** [comment]
3000 CJK punctuation: link in Katakana leads to term in Kanji. Is it right?
![image](https://github.com/user-attachments/assets/db17faf0-76d9-4f2a-8a1d-008266416d38)

**Comment by Mercury13** [comment]
Term “Status”: absolutely converse. Link in Kanji leads to term in Kana. Is it right?
![image](https://github.com/user-attachments/assets/058ab857-8309-465b-9460-5e4e50a10d4f)

**Comment by Mercury13** [comment]
31C0 CJK strokes, term in kana — right?
![image](https://github.com/user-attachments/assets/717540f6-58b0-4152-be6e-987de7affeab)

**Comment by Mercury13** [comment]
Katakana as script name is written with Kanji, elsewhere as KaTaKaNa.
![image](https://github.com/user-attachments/assets/d14b84b6-1b17-4f83-ba96-5d646ce4afde)
![image](https://github.com/user-attachments/assets/73c03d79-df51-4b54-bd32-e2d715a53739)

**Comment by Mercury13** [comment]
Same for Hiragana
![image](https://github.com/user-attachments/assets/30b93370-e8b2-4719-b02f-0e53c9d1d8df)

**Comment by Mercury13** [comment]
Just too lazy to kick those problems. Hell poor work, you probably auto-translated without even testing.

**Comment by coolvitto** [comment]
In Japanese, the distinction between kanji, katakana, and hiragana is subtle, and sometimes they are changed to suit the surrounding context.
I made the change with that in mind, but it would be better to standardize it (especially in the case of wikis).

I am using "pre-release 2.9.3.1".
Regarding sorting, I think it is well sorted in Japanese as well.

“2D80 Ethiopic ex” is missing brackets in the original.
![sshot-1](https://github.com/user-attachments/assets/e0997791-501f-45ce-b534-b623dabe456c)

12/22 fix-1
[Japanese_Fix_2024_12_22-Fix1.zip](https://github.com/user-attachments/files/18220805/Japanese_Fix_2024_12_22-Fix1.zip)

**Comment by Mercury13** [comment]
Are you ABSOLUTELY right?
You replaced native Ku:haku with borrowed SuPe:Su.
And dictionaries state only interstellar space, not empty space on paper.
![image](https://github.com/user-attachments/assets/03ee3e3e-fb11-4c97-bf47-38f81ba109a5)

**Comment by Mercury13** [comment]
Upd: Wikipedia states Su:PeSu, so you’re really right.

**Comment by Mercury13** [comment]
Hira here. TEST, TEST, bro!
![image](https://github.com/user-attachments/assets/ad1af945-5591-43dc-a1bb-c631c227f7d7)

**Comment by Mercury13** [comment]
Here’s what you have made.
![image](https://github.com/user-attachments/assets/4d13a68e-f53b-43e9-a177-8be84f65ea89)

**Comment by Mercury13** [comment]
IMPORTANT: Almost sure that no SuPeSu here.
Spacing character = character that has its own width and moves cursor at least a little.
![image](https://github.com/user-attachments/assets/6fe53bd3-9ed8-4af1-9e0a-8058dccbb186)

**Comment by Mercury13** [comment]
NEVER translate four-letter script codes. They are a link to something.

**Comment by coolvitto** [comment]
The reason why ([NO LANGUAGE!!!]) is displayed is because {{nspk|Hira}} is included.
![sshot-2](https://github.com/user-attachments/assets/fd9216e0-e8db-4b2f-b75c-1e7b48f6f427)

delete {{nspk|Hira}}, it will not be displayed.
![sshot-3](https://github.com/user-attachments/assets/e8406682-7cd2-4da2-9187-f8698281f79c)

Maybe {{nspk|Hira}} is not necessary?

**Comment by coolvitto** [comment]
```
IMPORTANT: Almost sure that no SuPeSu here.
Spacing character = character that has its own width and moves cursor at least a little.
```

Sorry.
I don't quite understand what this English means.

Is the translation of "Spacing character"
スペース文字
OK?

**Comment by Mercury13** [comment]
Probably NO.
1. Spacing = taking some room, NOT SuPeSu = character that has no graphic image.
2. Character = symbol. You probably translated English homonym character = traits of a human; some fictional person that’s involved in a work of fiction.

{{nspk|Hira}} is NEEDED, because # of people speaking Japanese is tied to Hiragana, and in Katakana

**Comment by Mercury13** [comment]
Probably character = 記号 Kigo, but IDK, I use Google Translate. You are the one who knows Japanese.

**Comment by Mercury13** [comment]
The same Kigo I see in Wikipedia in the beginning if Plus. But as Latin has the same 文字, you’re probably right.

**Comment by Mercury13** [comment]
So, the only remaining is Spacing — taking place. Not just SuPeSu = ␣.

**Comment by Mercury13** [comment]
I found Unicode glossary
https://www.unicode.org/terminology/term_ja_en.html

**Comment by coolvitto** [comment]
`https://www.unicode.org/terminology/term_ja_en.html`
I have already referred to this link.

Initially, I translated "Spacing character" as "間隔", but is this closer to the image?
"間隔" = A word that expresses the space between things
How about "Spacing character" = 間隔文字?

There is no such character in Japanese, so it is very difficult to translate.

**Comment by Mercury13** [comment]
No, not space between things. Use something like 前進を伴う.

**Comment by coolvitto** [comment]
「スペース文字」 or 「空白文字」
Which do you think is more appropriate?

**Comment by Mercury13** [comment]
NO, This is WHITESPACE character, e.g. character that only have width but no graphical representation.
Spacing are almost everything that’s visible on screen, including letters, digits, signs, whitespace, emoji — except control, format and some marks.

I see, this thing 前進を伴う is translated as “advancing character”, and that’s OK.

**Comment by coolvitto** [comment]
So, is 「空白文字」 okay?

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/c9990e94-35a4-4245-bdb2-1dc2a78c9774)
May be either SuPaSu or 空白, both a valid. Invent something around 前進を伴う (“advancing forward”).

**Comment by coolvitto** [comment]
Similarly, a 「空白文字」 refers to a character that involves forward movement of the current position, so for now we will translate it as 「空白文字」.
ok?

**Comment by Mercury13** [comment]
If OK then OK, you know the language — though I see that 空白 is blank.

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/5f0a5d1a-688a-46fa-8499-2d32fe688921)

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/575a2062-e36c-48be-afeb-4281d0bcdaa9)

**Comment by Mercury13** [comment]
Spacing character = everything that has a width. Maybe translate in that way?

**Comment by Mercury13** [comment]
In Russian it “has a length”.

**Comment by coolvitto** [comment]
The Japanese word 「空白」 also means a gap, so I think it can also be used in place of width (length).

**Comment by Mercury13** [comment]
No, it is NOT a gap, and 空白文字 Google clearly translate like “whitespace character”.
Reverso does too: https://context.reverso.net/translation/japanese-english/%E7%A9%BA%E7%99%BD%E6%96%87%E5%AD%97

**Comment by coolvitto** [comment]
Hmm, it's very difficult.

「長さ文字」 and 「前進文字」 are terms that Japanese people don't understand.
長さ = length
前進 = forward

**Comment by Mercury13** [comment]
Here’s invention of Unicode Consortium — as you see, 5 characters.
前進を伴う. IDK if it’s OK. But in Wikipedia…
![image](https://github.com/user-attachments/assets/2ade3a2c-4868-4894-acd2-488aae9d3f17)

**Comment by coolvitto** [comment]
By the way, AI translates Unicode's "Spacing character" as 「スペース文字」 or 「空白文字」.

**Comment by Mercury13** [comment]
Both are WHITESPACE characters.

**Comment by coolvitto** [comment]
In my opinion, 「前進を伴う」 is a description of the meaning of the words, not a description of the letters themselves.

**Comment by Mercury13** [comment]
I’ve also seen this
https://x0213.org/wiki/wiki.cgi?page=JIS+X+0201
![image](https://github.com/user-attachments/assets/616b5382-9a6c-405a-961c-b1f28aee99d2)

**Comment by coolvitto** [comment]
This is an explanation of the action of the character, not the name of the character.

Also, voiced/semi-voiced consonants and "spacing character" are different things.

**Comment by Mercury13** [comment]
Yes, it’s action. They are not names, it’s an extremely broad type of characters that spans letters, digits, signs, spaces, some marks etc.

**Comment by coolvitto** [comment]
This translation is difficult, so we'll reconsider it later.

For now, we'll translate it as 「空白文字」.

**Comment by Mercury13** [comment]
That’s what they mean.
![image](https://github.com/user-attachments/assets/8ec0d712-0aa0-454a-a575-c95f23e79150)
NOOOOOOOOOOOOOOOOO, it’s translated as WHITESPACE CHARACTER.

**Comment by Mercury13** [comment]
LEFT picture: 1 is spacing 2 is not. RIGHT: both are spacing.

**Comment by coolvitto** [comment]
So, should we leave the "スペース" as it is for now (until we can resolve this)?

**Comment by Mercury13** [comment]
No, I found that 前進を伴う文字 actually exists in sources as SPACING character.
スペース文字 and 空白文字 will also exist in Unicodia as WHITESPACE characters.

**Comment by Mercury13** [comment]
20 space and A0 NBSP are SPACING and WHITESPACE
ABC are just SPACING.

**Comment by Mercury13** [comment]
Read the description of Kirat Rai script (code Krai). Though it’s a Brahmic abugida, it does not contain marks on marks and other typographic difficulties, all characters are ordinary spacing.

**Comment by coolvitto** [comment]
So, is 「前進を伴う文字」 okay?

**Comment by Mercury13** [comment]
Should be okay, as it actually exists in sources.

**Comment by coolvitto** [comment]
Yes, I understand that "Spacing character" and "Whitespace" are different things.

However, 「前進を伴う文字」 is not understood by Japanese people. Is that okay?

**Comment by Mercury13** [comment]
Okay, it’s highly technical term. The only normal translation is in Ukrainian (Russian and Ukrainian are close relatives, but in Russian it’s worse)
Lao: many marks are SPACING for historical reasons
Arabic presentation forms: contain SPACING vowel marks
Seperator/whitespace: all SuPaSu (or 空白) are SPACING characters.
Virama: most viramas are combining. But some are SPACING.

**Comment by Mercury13** [comment]
BTW, block 02B0 is 前進を伴う修飾文字 in Wikipedia.

**Comment by coolvitto** [comment]
In my opinion, if it's a 「空白文字」, it will be possible to understand it through a search even if the meaning is slightly different.
However, if it's a 「前進を伴う文字」, I think it will be very difficult to understand through a search.

**Comment by Mercury13** [comment]
Search through 空白文字 returns spaces, this category of characters.
![image](https://github.com/user-attachments/assets/55fd16a7-c471-43a2-ad93-689042496bc3)

**Comment by Mercury13** [comment]
Japanese Wikipedia has exactly this term for WHITESPACE. https://ja.wikipedia.org/wiki/%E7%A9%BA%E7%99%BD%E6%96%87%E5%AD%97

**Comment by coolvitto** [comment]
So, instead of 「前進を伴う文字」, how about we call them 「前進を伴う修飾文字」?
This way, I think Japanese people would understand.

**Comment by Mercury13** [comment]
Invent something instead of Shūshoku (modifier).

**Comment by Mercury13** [comment]
![Uploading image.png…]()

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/860bf92e-24ca-4f4c-8fd3-b95dc90d649d)

**Comment by coolvitto** [comment]
「前進を伴う修飾字」ok?

**Comment by Mercury13** [comment]
But you haver LEFT 修飾, though I asked how to get rid of it and replace with something more neutral?

**Comment by coolvitto** [comment]
or
「前進用修飾字」

**Comment by Mercury13** [comment]
前進用文字 Is it OK?

**Comment by Mercury13** [comment]
修飾 has special meaning in Unicode (though rare, mostly super/subscript letters)

**Comment by coolvitto** [comment]
ok
Let's decide on "前進用文字"

**Comment by coolvitto** [comment]
I don't really understand the difference between "pt:space|space" and "pt:spacing|spacing".

pt:space|space = pt:space|スペース
pt:spacing|spacing = pt:spacing|前進用
ok?

**Comment by Mercury13** [comment]
Yes. Space = character that has only width, no graphic representation
Spacing = has width, see picture above: 2 left has no width and is thus non-spacing.

**Comment by coolvitto** [comment]
thanks

**Comment by coolvitto** [comment]
Regarding the explanation of whitespace,
![名称未設定-1](https://github.com/user-attachments/assets/29129c1a-aa6d-4f63-8a81-7873124b0ee6)

it all shows "146%___", is this normal?

Is it because I'm missing some font?

**Comment by coolvitto** [comment]
About the current icon
![icon2](https://github.com/user-attachments/assets/544b0ae7-71fb-4ab9-a7e8-80ce17ec36eb)


I don't really understand the point of using voiced marks in the icon, but how about the following?
![icon1](https://github.com/user-attachments/assets/edbd8efc-4801-4cca-8585-6d16671b74d2)

カ = KA
ク = KU
あ = A

**Comment by Mercury13** [comment]
About 146% (a common Russian meme about allegedly botched elections). Different spaces. The last is 1px short in your font, but is nevertheless the best of all three.
Hira A — I’ll do if you suggest it’s better. It was just a random sign.
Kana KA — I’ll search among other signs.
Kana KU — does it exist in small? DOES. I’ll check how it looks in a half-cell.

**Comment by Mercury13** [comment]
Well, I looked at the picture in whole, and voiced marks appeared to be a good telltales for kana.
![image](https://github.com/user-attachments/assets/538ce849-46c3-4f75-be6d-2bef9518fc79)

**Comment by Mercury13** [comment]
I don’t remember why I made these icons 3 years ago, but an important consideration was GIMP’s hinting. GIMP has more options for lo-fi text display than Photoshop, and is just free. In winter 2022/23 I switched to SVG for HiDPI, mostly because I actually tested Unicodia on HiDPI laptops, and then bought mine at disassembly.

**Comment by coolvitto** [comment]
So that's the reason why they use voiced consonants.
However, Japanese people don't normally use characters with voiced consonants such as "ji."

I'm sorry for the trouble, but I'd be grateful if you could create it based on my suggestion.

**Comment by Mercury13** [comment]
Then which voiced ARE used?

**Comment by Mercury13** [comment]
In autumn 2022 I had such a laptop for a few nights, and just fixed everything HiDPI-related except icons. Then I decided to translate to Ukrainian, and new icons were signs that this part of work is done. Ukrainian and HiDPI went in parallel, they were hard jobs.

**Comment by coolvitto** [comment]
The most commonly used letter is "あ".
Other than that, the first letter of the item is often used.

**Comment by Mercury13** [comment]
Blackouts became so evil that I felt sick one day because of smoke from generators.

**Comment by coolvitto** [comment]
I also usually use a laptop as my main PC. I only use a desktop computer when I need to edit on a large screen.

**Comment by Mercury13** [comment]
Maybe make different icons for Japanese and the rest?

**Comment by coolvitto** [comment]
No, I only want Japanese icons.

**Comment by Mercury13** [comment]
I mean make in Japanese L10n special icons for kana, distinct from English/Russian?

**Comment by Mercury13** [comment]
Katakana small KU — accepted but I’ll check. It mostly consists of sloped lines, and as our SVG renderer does not have gamma correction, in lo-fi it’ll probably look really poor.
Katakana GA — OK if I manage to make it right. KA — IDK, I’ll check.
Hiragana A — that’s your sign for Hiragana, right?

**Comment by coolvitto** [comment]
Yes

**Comment by coolvitto** [comment]
If you're okay with the "146%___" notation, then this is the corrected version for now.
[Japanese_Fix_2024_12_23.zip](https://github.com/user-attachments/files/18229658/Japanese_Fix_2024_12_23.zip)

**Comment by Mercury13** [comment]
Need to run.
nspk|Hira. The rest accepted.
![image](https://github.com/user-attachments/assets/a6605db5-2af3-48f2-b78f-f483eb44da73)

**Comment by coolvitto** [comment]
```
Need to run.
nspk|Hira. The rest accepted.
```
Sorry.
I don't quite understand the meaning.

{{nspk|ひらがな}} - > NG
{{nspk|Hira}} - > OK ?

**Comment by coolvitto** [comment]
ok, solved.
[Japanese_Fix_2024_12_23_fix1.zip](https://github.com/user-attachments/files/18229876/Japanese_Fix_2024_12_23_fix1.zip)

**Comment by Mercury13** [comment]
{{nspk|Hira}} - > OK ?
Parameter 1 is code of language or script. Japanese is coded under script Hira.

**Comment by Mercury13** [comment]
And Marathi is coded under script Modi, though the main for it is Devanagari.

**Comment by coolvitto** [comment]
You are truly knowledgeable. 
I feel so lucky to have met you.

**Comment by Mercury13** [comment]
Sorry, did not work at all. Four-letter script codes are not translated.

**Comment by Mercury13** [comment]
They are TECHNICAL codes. ps:Hira = pop up, script, Hiragana.

**Comment by Mercury13** [comment]
Probably I’ll stick together Hiragana and Katakana. Is it cool?

**Comment by coolvitto** [comment]
Do you mean a general term for both hiragana and katakana?
If so, I don't think there's any problem.

**Comment by Mercury13** [comment]
No.
• Hiragana
• …Katakana
• …Kana phonetic extensions
• …Some more small kana

**Comment by Mercury13** [comment]
Regardless of localization, the order will remain the same, and only ellipses are prepended if wrong order.

**Comment by Mercury13** [comment]
Maybe this will compensate troubles with scanning for kana.

**Comment by coolvitto** [comment]
Oh, I think that's a pretty good idea.

**Comment by Mercury13** [comment]
Merry Christmas!
Unicodia’s logo contains “Unicodia” in different scripts of different continents. As the name is not native Japanese, I had no moral right to draw “Yu” in Hiragana.
![image](https://github.com/user-attachments/assets/e86c535e-7889-491d-a747-178d7410ed09)

**Comment by coolvitto** [comment]
Merry Christmas!

Wow!!
The logo was in Unicode!!!!!
You're amazing as always.

**Comment by Mercury13** [comment]
Hiragana A is here and did not really affect scanning. There was a blackout in the morning, just partly within enemy’s modus operandi.
![image](https://github.com/user-attachments/assets/a2d724fe-ffc2-43f7-98ad-895be79a3250)

**Comment by Mercury13** [comment]
I’ll think what to put to Katakana, KA or GA.

**Comment by coolvitto** [comment]
Wow!! Thank you so much!!
You have great aesthetic sense too!

**Comment by Mercury13** [comment]
That’s how it looks now. Katakana phonetic is not really cool, but “within error”. Accepted.
![image](https://github.com/user-attachments/assets/072d5e17-1342-482e-b6cc-d2f921556fd7)

**Comment by coolvitto** [comment]
Great!!
Thank you so much.

**Comment by Mercury13** [comment]
Splitting Asia into two clusters: RtL and LtR.
![image](https://github.com/user-attachments/assets/9a755591-76ff-4fb6-a4a7-e3c539ee83e2)

**Comment by coolvitto** [comment]
I see, I think that’s a very good thing.

**Comment by coolvitto** [comment]
By the way, I will be away from home for about 10 days starting tomorrow.

I know it's a little early, but I wish you a happy new year.

**Comment by Mercury13** [comment]
Thank you. Probably releasing as is, but first need to invent style for Asia ←. Hard.

**Comment by coolvitto** [comment]
What is the Asian style?

I think it's great as it is...

**Comment by coolvitto** [comment]
Ah, I see!
You mean LtR and RtL, right?

**Comment by Mercury13** [comment]
Yes. Left-to-right and right-to-left. At least some formal sign.

**Comment by Mercury13** [comment]
Mongolian and Phags-Pa are still ambiguous, but let them be LtR.

**Comment by Mercury13** [comment]
![image](https://github.com/user-attachments/assets/f8b03fa1-552e-4585-8d07-f295239e35a2)
Does this phenomenon exist in Japanese? Is kataKana/hiraGana very this phenomenon?

**Comment by Mercury13** [comment]
CJK ideographs. Here I’m not checking you, instead you are checking me. Have I written correctly about onyomi and kunyomi?
![image](https://github.com/user-attachments/assets/d85d0f37-4de8-419c-b4d3-1ff9017be672)

**Comment by Mercury13** [comment]
Once again about sandhi: AFAIK, np and nb is pronounced [mp] and [mb]. If I’m true, it’s very this phenomenon.

**Comment by Mercury13** [comment]
I don’t know Japanese, but have invented a mnemonic rule about onyomi and kunyomi: “kun” is friend → Japanese. “On” in Russian is “he” → Chinese.

**Comment by Mercury13** [comment]
Kharoshthi:
![image](https://github.com/user-attachments/assets/2ed79661-10f0-408b-97f0-abe7bcb67884)

**Comment by Mercury13** [comment]
Do I need  to internationalize these?
![image](https://github.com/user-attachments/assets/81c246f0-10fa-4381-8338-7bc1cdb16d76)
![image](https://github.com/user-attachments/assets/e4c0354d-3de8-4b8f-aad7-2b8c4a7680cd)

**Comment by Mercury13** [comment]
Tai Tham: may or may not be mistake, please check
(in Russian/Ukrainian I translated “high”)
![image](https://github.com/user-attachments/assets/bf3b7d15-ffb3-4d9b-a26c-84fb5e651375)

**Comment by Mercury13** [comment]
Balinese: untranslated link
![image](https://github.com/user-attachments/assets/f653da25-ab68-4274-89cb-36e17775959f)

**Comment by Mercury13** [comment]
Number forms
![image](https://github.com/user-attachments/assets/e3b170be-58cc-45ef-9831-c55cdeef204f)

Does it work in YOUR system? I’ll try to improve.

**Comment by Mercury13** [comment]
Number forms 2150: need European dash here?
![image](https://github.com/user-attachments/assets/bdfbade4-55e6-4b2a-88b0-f4380f34e934)

**Comment by Mercury13** [comment]
Do you mind finding and replacing apostrophes to U+2019, or you got issues in your font?

**Comment by Mercury13** [comment]
Saur Saurashtra: English term OK?
![image](https://github.com/user-attachments/assets/9d36a85a-33e0-4ef4-9b4d-eb98e0d8de18)

**Comment by Mercury13** [comment]
Rejang: same

**Comment by Mercury13** [comment]
Inconsistencies in Ethiopic:
2D80
![image](https://github.com/user-attachments/assets/eeba310d-7db5-4b03-a936-156a0c96553a)

AB00
![image](https://github.com/user-attachments/assets/428de9a3-0e65-48ad-87ef-3a68f8696083)

**Comment by Mercury13** [comment]
10190 ancient symbols
Is “cross/Christ” Greek language? Then why are you leaving them in Latin?
![image](https://github.com/user-attachments/assets/f930ac90-3723-48cd-95f8-cd9ab1655b21)

**Comment by Mercury13** [comment]
101D0 Phaistos Disc: Latin w/o justification
![image](https://github.com/user-attachments/assets/0a6a3388-cacd-4891-b9e6-9aa19d16ddc0)

**Comment by Mercury13** [comment]
Perm Old Permic
![image](https://github.com/user-attachments/assets/d73b2e54-c103-4472-ba9e-7565100a9d14)
Is “Anbur” needed here?

**Comment by Mercury13** [comment]
Elba Elbasan: same, Latin w/o justification
![image](https://github.com/user-attachments/assets/2f53c2e5-d9d3-49c2-9473-cdf6a9c4216b)

**Comment by Mercury13** [comment]
Multiple: does italic ever exists as font style in Japanese?

**Comment by Mercury13** [comment]
Sidd Siddham: Does it need Latin names?
![image](https://github.com/user-attachments/assets/cf6399b3-7f4b-4349-9924-ce86c0da6493)

**Comment by Mercury13** [comment]
Shrd Sharada: does it need Latin?
![image](https://github.com/user-attachments/assets/1fa52353-3c01-4f20-913b-140cf2641487)

**Comment by Mercury13** [comment]
Diak Dives Akuru: Latin here?
![image](https://github.com/user-attachments/assets/0497137c-e48b-457c-bbe9-794cfb1e622f)

**Comment by Mercury13** [comment]
Nand Nandinagari: both Japanese in samples and Latin in links
![image](https://github.com/user-attachments/assets/f6fe780b-eef7-45b5-9220-b5df30e643f3)

**Comment by Mercury13** [comment]
Bhks Bhaiksuki: same
![image](https://github.com/user-attachments/assets/08cb97e6-dabf-4457-8ab6-75d610f7d5e8)

**Comment by Mercury13** [comment]
Maka Makasar: Latin in links
![image](https://github.com/user-attachments/assets/c30f2881-f688-441d-a6cb-c8f91dcbe691)

**Comment by Mercury13** [comment]
18D00 Tangut ex: Latin w/o justification
![image](https://github.com/user-attachments/assets/a420efa0-4b0d-4c1e-a39b-2d679c774c94)

**Comment by Mercury13** [comment]
Onao Ol Onal: Is “danda” needed in Latin?
![image](https://github.com/user-attachments/assets/6f96cec6-ddbc-4f93-a545-173280b167c6)

Ottoman Siyaq 1ED00: Latin in link (the rest, of course, you can close somehow too, but not really a mistake)
![image](https://github.com/user-attachments/assets/3b9b7939-084f-4d67-b707-b4447e4e09d0)

**Comment by Mercury13** [comment]
1F650 ornamental dingbats: Is term “dingbat” needed?
![image](https://github.com/user-attachments/assets/7be5ce60-5a35-4964-8ebd-343f604cb5e6)

**Comment by Mercury13** [comment]
I wrote L10n of punctuation, in locale.xml.
	
	<punctuation key-value-colon=": " uniform-comma=", " semicolon="; " />
What are Japanese counterparts?

**Comment by Mercury13** [comment]
I wrote L10n of punctuation.
```
	<!-- Punctuation
			key-value-colon: in “UTF-8: 00”
			uniform-comma: in uniform words “first, second”
			semicolon: in “first, second; did the third”
				(script’s languages only, in English see Sunuwar)
			left-paren, right-paren: in script’s languages only -->
	<punctuation key-value-colon=": " uniform-comma=", " semicolon="; "
			left-paren="(" right-paren=")" />
```
Which are Japanese counterparts?

**Comment by Mercury13** [comment]
For now I took: fullwidth colon FF1A  + space 20; ideo comma 3001; fullwidth semicolon FF1B; fullwidth parentheses FF08/09.

**Comment by Mercury13** [comment]
Repeat question: is italic OK in Japanese, or disable it specially for Japanese?

**Comment by coolvitto** [comment]
Happy New Year!!

I'm back.

It looks like you've updated to v2.9.4.
Thank you for your hard work.

I can't open the "Japanese language resources" thread on Github.

I've confirmed many of the issues through email, but I have a full-time job starting tomorrow, so it may be difficult to address all of the issues.


>> is italic OK in Japanese, or disable it specially for Japanese?
When you want to emphasize something in Japanese, you can use italics. As for font decoration, you don't need to pay special attention to Japanese, so I think it's fine to use it the same as other languages.
________________________________
差出人: Mercury13 ***@***.***>
送信日時: 2025年1月5日 7:30
宛先: Mercury13/unicodia ***@***.***>
CC: coolvitto ***@***.***>; Mention ***@***.***>
件名: Re: [Mercury13/unicodia] Japanese language resources (Issue #424)


Repeat question: is italic OK in Japanese, or disable it specially for Japanese?

**Comment by Mercury13** [comment]
Of course, do it as slow as you wish. I added a little more strings.

**Comment by Mercury13** [comment]
If you update *.utran, I addressed some of the issues there. But I cannot address all, of course.

**Comment by Mercury13** [comment]
I’m working on L10n task for Unicode 17. Approvals are incomplete, and maybe there’ll be more blocks.

**Comment by Mercury13** [comment]
@coolvitto Could you do some translation? Even two strings a week is OK.

**Comment by Mercury13** [comment]
I’m waiting for January meeting. Probably they’ll release U17 alpha.

**Comment by coolvitto** [comment]
I'm currently busy with other work, so I'll help with the translation when I have some free time.
I think I'll have a bit more free time in February, so please wait a little longer.
I hope to continue helping with Unicodia translations in the future.

**Comment by Mercury13** [comment]
Anyway, I made a task for you in advance. Everything I know about U17 is already in language resources.

**Comment by Mercury13** [comment]
@coolvitto As you see, translation to Chinese failed because of accident at the other side.
I auto-translated a few strings to Japanese, and marked them as bad.
E.g. Sidetic = Shide (Katakana) Moji (ideographs).

**Comment by Mercury13** [comment]
@coolvitto 
I’ve recently seen really strange rules of writing # of native speakers in Japanese. Could you explain them to me?

**Comment by Mercury13** [comment]
@coolvitto See #475

**Comment by Mercury13** [comment]
Jindai moji (神代文字) is your Japanese fake. Why write with Latin script?

**Comment by Mercury13** [comment]
@coolvitto Are you here?
